modules/videoio/src/cap_v4l.cpp

```diff
@@ -285,71 +284,61 @@ struct CvCaptureCAM_V4L CV_FINAL : public CvCapture
     bool convert_rgb;
     bool frame_allocated;
     bool returnFrame;
+    int channelNumber;
 
     /* V4L2 variables */
     buffer buffers[MAX_V4L_BUFFERS + 1];
-    v4l2_capability cap;
-    v4l2_input inp;
+    v4l2_capability capability;
+    v4l2_input videoInput;
     v4l2_format form;
-    v4l2_crop crop;
-    v4l2_cropcap cropcap;
     v4l2_requestbuffers req;
     v4l2_buf_type type;
-    v4l2_queryctrl queryctrl;
 
     timeval timestamp;
 
-    /* V4L2 control variables */
-    Range focus, brightness, contrast, saturation, hue, gain, exposure;
-
     bool open(int _index);
     bool open(const char* deviceName);
+    bool isOpened() const;
 
     virtual double getProperty(int) const CV_OVERRIDE;
     virtual bool setProperty(int, double) CV_OVERRIDE;
     virtual bool grabFrame() CV_OVERRIDE;
     virtual IplImage* retrieveFrame(int) CV_OVERRIDE;
 
-    Range getRange(int property_id) const {
-        switch (property_id) {
-        case CV_CAP_PROP_BRIGHTNESS:
-            return brightness;
-        case CV_CAP_PROP_CONTRAST:
-            return contrast;
-        case CV_CAP_PROP_SATURATION:
-            return saturation;
-        case CV_CAP_PROP_HUE:
-            return hue;
-        case CV_CAP_PROP_GAIN:
-            return gain;
-        case CV_CAP_PROP_EXPOSURE:
-            return exposure;
-        case CV_CAP_PROP_FOCUS:
-            return focus;
-        case CV_CAP_PROP_AUTOFOCUS:
-            return Range(0, 1);
-        case CV_CAP_PROP_AUTO_EXPOSURE:
-            return Range(0, 4);
-        default:
-            return Range(0, 255);
-        }
-    }
-
+    CvCaptureCAM_V4L();
```

Comment:
as you are rewriting this anyway, you can consider porting this to the new `IVideoCapture` API

Reply:
unless you actually need this in 3.4 that is..

Reply:
but then again you should leave parameter conversion as is - upgrading to 3.4.y should not needlessly break things

Reply:
I remove parameter conversation because I can't set/get some of it in this case. I got INF for exposure. Look at stackoverflow - working with opencv-v4l is a pain. Most of parameters a missed. So, why you  want to leave  senseless parameter conversion when it so hard to use?
If you like, I rebase this to 4.0. Again, my goal wasn't rewrite v4l, I want to work with the camera in the same way on both linux and windows: the same parameters and the same synchronization.

Reply:
> So, why you want to leave senseless parameter conversion when it so hard to use?

backwards compatibility. There is working code that provides the parameters in the senseless format and it would break by this improvement.

Breaking on 4.0 should be fine though.

> Look at stackoverflow - working with opencv-v4l is a pain.

that is a little bit unspecific..

Reply:
ok, I will rebase it on 4.0 branch. Should I close this PR?

Reply:
You can rebase this patch "inplace". No need to reopen PR.